### ISSUE TITLE
Alter update with materialised column depending on ephemeral is not working.

### ISSUE THREAD

**Issue by iyuroch** [ClickHouse/ClickHouse#68839]
**Company or project name**

Perfectscale

**Describe what's wrong**

Alter table update with materialized column that depends on ephemeral is returning exception.

https://fiddle.clickhouse.com/385d3c58-071f-4f3a-b0b6-f1ae0e1f20a1

**Does it reproduce on the most recent release?**

Yes, in fiddle

**How to reproduce**

Provided fiddle

**Expected behavior**

I expected alter update to work.

**Comment by max-vostrikov** [comment]
I would rephrase it. It makes sense that alter table will not work with ephemeral columns - the ephemeral columns do not exists.
What you expect to happen, is to treat ephemeral column as NULL during ALTER TABLE statement.
I see it as a new feature rather than a bug.

**Comment by iyuroch** [comment]
@max-vostrikov This also affects lightweight deletes which do not work with ephemeral columns tables. I think it's a bug due to this reason

**Comment by max-vostrikov** [comment]
Yes, added a delete example in fiddle [https://fiddle.clickhouse.com/55b0bfed-4832-40c5-b2d6-f4e5516b6e86](https://fiddle.clickhouse.com/55b0bfed-4832-40c5-b2d6-f4e5516b6e86).

I mean both the documentation and the error message is clear enough that ephemeral columns do not exists. But does look like it should be possible to treat them as NULL instead.

**Comment by iyuroch** [comment]
@max-vostrikov Yes it does not exist, however, I believe that it should be irrelevant for lightweight deletion thus a bug

**Comment by kurtraschke** [comment]
This issue and #71793 still exist on current releases.  Is there any update on this issue? It should, at minimum, be possible to use `DELETE FROM` and `ALTER TABLE ... UPDATE` on a table with `EPHEMERAL` columns **with a condition that does not reference the `EPHEMERAL` column**.

(To clarify, as the scope is larger than the title of the bug suggests: right now it is **not possible** to use `DELETE FROM` or `ALTER TABLE ... UPDATE` on a table with an `EPHEMERAL` column _even if_ the condition does not reference the `EPHEMERAL` column or any columns whose values depend on it.)
